rust/lib.rs

```diff
@@ -203,6 +237,49 @@ where
             Some(unsafe { result.offset_from(haystack_pointer) } as usize)
         }
     }
+
+    fn sz_edit_distance(&self, needle: N) -> usize {
+        let haystack_ref = self.as_ref();
+        let needle_ref = needle.as_ref();
+        let haystack_length = haystack_ref.len();
+        let needle_length = needle_ref.len();
+        let haystack_pointer = haystack_ref.as_ptr() as _;
+        let needle_pointer = needle_ref.as_ptr() as _;
+        unsafe {
+            sz_edit_distance(
+                haystack_pointer,
+                haystack_length,
+                needle_pointer,
+                needle_length,
+                // Upper bound on the distance, that allows us to exit early. If zero is
+                // passed, the maximum possible distance will be equal to the length of
+                // the longer input.
+                0,
+                // Uses the default allocator
+                core::ptr::null(),
+            )
+        }
+    }
+
+    fn sz_alignment_score(&self, needle: N, matrix: &mut [[i8; 256]; 256], gap: i8) -> isize {
+        let haystack_ref = self.as_ref();
+        let needle_ref = needle.as_ref();
+        let haystack_length = haystack_ref.len();
+        let needle_length = needle_ref.len();
+        let haystack_pointer = haystack_ref.as_ptr() as _;
+        let needle_pointer = needle_ref.as_ptr() as _;
+        unsafe {
+            sz_alignment_score(
+                haystack_pointer,
+                haystack_length,
+                needle_pointer,
+                needle_length,
+                matrix.as_mut_ptr() as _,
```

Comment:
I believe this is an undefined behavior. Is it guaranteed, that `[[i8; 256]; 256]` forms a continuous block of memory?